jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/CrossOriginHandler.java

```diff
@@ -0,0 +1,423 @@
+        {
+            origin = origin.trim();
+            if (origin.isEmpty())
+                continue;
+            for (Pattern pattern : allowedOriginPatterns)
+            {
+                if (pattern.matcher(origin).matches())
+                    return true;
+            }
+        }
+        return false;
+    }
+
+    private boolean isPreflight(Request request)
+    {
+        return HttpMethod.OPTIONS.is(request.getMethod()) && request.getHeaders().contains(HttpHeader.ACCESS_CONTROL_REQUEST_METHOD);
+    }
+
+    private void handlePreflightResponse(String origin, Response response)
+    {
+        HttpFields.Mutable headers = response.getHeaders();
+        headers.put(HttpHeader.ACCESS_CONTROL_ALLOW_ORIGIN, origin);
+        if (isAllowCredentials())
+            headers.put(HttpHeader.ACCESS_CONTROL_ALLOW_CREDENTIALS, "true");
+        Set<String> allowedMethods = getAllowedMethods();
+        if (!allowedMethods.isEmpty())
+            headers.put(HttpHeader.ACCESS_CONTROL_ALLOW_METHODS, String.join(",", allowedMethods));
+        Set<String> allowedHeaders = getAllowedHeaders();
+        if (!allowedHeaders.isEmpty())
+            headers.put(HttpHeader.ACCESS_CONTROL_ALLOW_HEADERS, String.join(",", allowedHeaders));
+        long seconds = getPreflightMaxAge().toSeconds();
+        if (seconds > 0)
+            headers.put(HttpHeader.ACCESS_CONTROL_MAX_AGE, seconds);
+    }
+
+    private void handleSimpleResponse(String origin, Response response)
+    {
+        HttpFields.Mutable headers = response.getHeaders();
+        headers.put(HttpHeader.ACCESS_CONTROL_ALLOW_ORIGIN, origin);
+        if (isAllowCredentials())
+            headers.put(HttpHeader.ACCESS_CONTROL_ALLOW_CREDENTIALS, "true");
+        Set<String> exposedHeaders = getExposedHeaders();
+        if (!exposedHeaders.isEmpty())
+            headers.put(HttpHeader.ACCESS_CONTROL_EXPOSE_HEADERS, String.join(",", exposedHeaders));
+    }
+
+    private void resolveAllowedOrigins()
+    {
+        for (String allowedOrigin : getAllowedOriginPatterns())
+        {
+            allowedOrigin = allowedOrigin.trim();
+            if (allowedOrigin.isEmpty())
+                continue;
+
+            if ("*".equals(allowedOrigin))
+            {
+                anyOriginAllowed = true;
+                return;
+            }
+
+            allowedOriginPatterns.add(Pattern.compile(allowedOrigin));
+        }
+    }
+
+    private void resolveAllowedTimingOrigins()
+    {
+        for (String allowedTimingOrigin : getAllowedTimingOriginPatterns())
+        {
+            allowedTimingOrigin = allowedTimingOrigin.trim();
+            if (allowedTimingOrigin.isEmpty())
+                continue;
+
+            if ("*".equals(allowedTimingOrigin))
+            {
+                anyTimingOriginAllowed = true;
+                return;
+            }
+
+            allowedTimingOriginPatterns.add(Pattern.compile(allowedTimingOrigin));
```

Comment:
Should the case insensitive option be used here?
```suggestion
            allowedTimingOriginPatterns.add(Pattern.compile(allowedTimingOrigin, Pattern.CASE_INSENSITIVE));
```

Reply:
This has not been addressed.